numpy/core/arrayprint.py

```diff
@@ -455,11 +446,27 @@ def array2string(a, max_line_width=None, precision=None,
 
     """
 
+    if max_line_width is None:
+        max_line_width = _line_width
+
+    if precision is None:
+        precision = _float_output_precision
+
+    if suppress_small is None:
+        suppress_small = _float_output_suppress_small
+
+    if formatter is None:
+        formatter = _formatter
+
     if a.shape == ():
         x = a.item()
-        if isinstance(x, tuple):
-            x = _convert_arrays(x)
```

Comment:
This is one of the two places `_convert_arrays` is called, which is the old-style structure formatter. Is the other call (in `_formatArray`) still needed? It looks like that one can never be reached. If so, you might remove `_convert_arrays` completely.